lib/rihanna.ex

```diff
@@ -91,6 +93,25 @@ defmodule Rihanna do
     raise ArgumentError, @enqueue_help_message
   end
 
+  @spec enqueue_with_connection({module, atom, list()}, connection) :: {:ok, Rihanna.Job.t()}
+  def enqueue_with_connection(term = {mod, fun, args}, connection)
+      when is_atom(mod) and is_atom(fun) and is_list(args) do
+    Rihanna.Job.enqueue(term, producer_postgres_connection: connection)
```

Comment:
Yeah, just put it in the opts

Reply:
@tpitale sorry for the late reply, somehow I missed the update.

That was my first approach, as detailed in my PR comments:

> went initially for the optional argument in enqueue , but considering we have the same function with 2 or 3 arguments and one is optional, I get compilation error
    == Compilation error in file lib/rihanna.ex ==
    ** (CompileError) lib/rihanna.ex:91: def enqueue/3 defaults conflicts with enqueue/2
        lib/rihanna.ex:91: (module)

so something like this:
``` elixir 
def enqueue(term , opts \\ %{})
def enqueue(mod, arg, opts \\ %{})
```
as `enqueue/2` can match both of the above.

Or did you had some other suggestion?

Reply:
I was thinking that there would be a new enqueue for a two-element `term` tuple, that would support the `opts`. And the existing `enqueue/2` for `mod` + `arg` would remain for backwards compatibility. That would _not_ support `opts`, but if someone wanted to use `opts` they could just pass a 2-element tuple. This would kinda standardize with what `Rihanna.Job.enqueue` takes, which is just a `term` of 2 or 3 element tuple.

This is might look like (skipping comments):

```
  @spec enqueue({module, atom, list()}, map()) :: {:ok, Rihanna.Job.t()}
  def enqueue(term, opts \\ %{})

  def enqueue(term = {mod, fun, args}, opts)
      when is_atom(mod) and is_atom(fun) and is_list(args) do
    Rihanna.Job.enqueue(term, opts)
  end

  def enqueue(term = {mod, _args}, opts) when is_atom(mod) do
    Rihanna.Job.enqueue(term, opts)
  end

  @spec enqueue(module, any) :: {:ok, Rihanna.Job.t()}
  def enqueue(mod, arg) when is_atom(mod) do
    enqueue({mod, arg})
  end

  def enqueue(_, _) do
    raise ArgumentError, @enqueue_help_message
  end
```

Reply:
This is, of course, entirely up to @samphilipd for design approval.

Reply:
Thanks, makes sense to me. Ill try to put together a complete PR next week, then we can change it if needed.

Reply:
Updated the PR

Reply:
Seems fine to me, can you make sure the README and docs are updated?